packages/core/src/server/socketServer.ts

```diff
@@ -166,6 +172,29 @@ export class SocketServer {
       return null;
     }
 
+    // web-infra-dev/rspack#6633
+    // when initial-chunks change, reload the page
+    // e.g: ['index.js'] -> ['index.js', 'lib-polyfill.js']
+    const newEntryPointsChunks: string[] = [];
+    for (const entrypoint of Object.values(stats.entrypoints ?? {})) {
+      if (Array.isArray(entrypoint?.chunks)) {
+        const chunks = entrypoint?.chunks.filter(Boolean) as string[];
+        newEntryPointsChunks.push(...chunks);
+      }
+    }
+    newEntryPointsChunks.sort((a, b) => a.localeCompare(b));
```

Comment:
Can we use `Set()` to store these chunks?  I guess it will have better perf

1. Can dedupe
2. Can use `Set().has()` to check instead of sorting the array